packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -80,22 +77,37 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
     }
 
     private fun onConfigure(flutterResult: Result, request: HashMap<String, Any>) {
-        var modelSchemas: List<Map<String, Any>>
-        if (request.containsKey("modelSchemas") && request["modelSchemas"] is List<*>) {
-            modelSchemas = request["modelSchemas"].safeCastToList()!!
-        } else {
+
+        if( !request.containsKey("modelSchemas") || !request.containsKey("modelProviderVersion") || request["modelSchemas"] !is List<*>){
             prepareError(flutterResult,
-                         Exception(
-                                 FlutterDataStoreFailureMessage.AMPLIFY_QUERY_REQUEST_MALFORMED.toString()),
-                         FlutterDataStoreFailureMessage.ERROR_CASTING_INPUT_IN_PLATFORM_CODE.toString())
+                    Exception(
+                            FlutterDataStoreFailureMessage.AMPLIFY_QUERY_REQUEST_MALFORMED.toString()),
+                    FlutterDataStoreFailureMessage.ERROR_CASTING_INPUT_IN_PLATFORM_CODE.toString())
```

Comment:
We should fix this and use the correct error code.

Reply:
thanks my bad, using AMPLIFY_CONFIGURE_REQUEST_MALFORMED now